lib/spack/spack/cmd/ci.py

```diff
@@ -76,23 +76,40 @@ def setup_parser(subparser):
         default=False,
```

Comment:
^^ --optimized could probably be removed now right? It has been deprecated for a while.

Reply:
The deprecating [PR](https://github.com/spack/spack/pull/45005) says they'll be removed in `v0.24`, does that mean I can remove them now?

Reply:
Yeah I think so, removed.